### PR TITLE
Move stylelint to 14

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ <skjnldsv@protonmail.com>

### CODE REVIEW

**Comment by juliushaertl** [comment]
Right, there seems to be some additional things needed for properly linting (s)css on vue files again:

https://github.com/stylelint/stylelint/blob/14.0.0/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax

I couldn't get it to work in a quick attempt :/

**Comment by skjnldsv** [comment]
Should be good, please review

**Comment by skjnldsv** [comment]
> ERROR in Could not find "stylelint-config-recommended-vue". Do you need a `configBasedir`?

Oups

**Comment by artonge** [comment]
Added `postcss` to dependencies to make CI pass. Not sure if that is the best approach

**Review by artonge — APPROVED** [review]

**Comment by juliushaertl** [comment]
Right, i think it would make sense to rather add that to https://github.com/nextcloud/stylelint-config then, seems rather to be a peer dependency of that config

**Comment by artonge** [comment]
> Right, i think it would make sense to rather add that to https://github.com/nextcloud/stylelint-config then, seems rather to be a peer dependency of that config

The thing is, that this is this repo that needs it. `@nextcloud/stylelint-config` already has it as a dependency:

```bash

$ npm why postcss-html
> npm ERR! No dependencies found matching postcss-html

$ cd tests/

$ npm why postcss-html
> postcss-html@1.2.0 dev peer
node_modules/postcss-html
  peer postcss-html@"^1.0.0" from stylelint-config-html@1.0.0
  node_modules/stylelint-config-html
    stylelint-config-html@">=1.0.0" from stylelint-config-recommended-vue@1.0.0
    node_modules/stylelint-config-recommended-vue
      peer stylelint-config-recommended-vue@"^1.0.0" from @nextcloud/stylelint-config@2.0.1
      node_modules/@nextcloud/stylelint-config
        dev @nextcloud/stylelint-config@"^2.0.1" from the root project
  peer postcss-html@"^1.0.0" from stylelint-config-recommended-vue@1.0.0
  node_modules/stylelint-config-recommended-vue
    peer stylelint-config-recommended-vue@"^1.0.0" from @nextcloud/stylelint-config@2.0.1
    node_modules/@nextcloud/stylelint-config
      dev @nextcloud/stylelint-config@"^2.0.1" from the root project
```

In a normal scenario, I think that the tests would pass, but the fact the we use `"@nextcloud/webpack-vue-config": "file:.."`  [here](https://github.com/nextcloud/webpack-vue-config/blob/master/tests/package.json#L22), "splits" the `node_modules` folder in two, one in the root (`/`) folder and one in `tests`, which lead to `webpack` not finding the `postcss-html` package.

**Comment by artonge** [comment]
Attempting to use `npm pack`

**Comment by artonge** [comment]
CI is green when using `npm pack` and installing the resulting archive.

**Comment by artonge** [comment]
@skjnldsv An opinion on the new `npm-test.yml` workflow?

**Comment by skjnldsv** [comment]
> @skjnldsv An opinion on the new `npm-test.yml` workflow?

Yeah, looks nice!
I figured `"@nextcloud/webpack-vue-config": "file:.."` looked broken! :confused: 

Your change could use some comments to ensure anyone understand why we're doing this :)

File: .github/workflows/npm-test.yml [hunk 1]
```diff
@@ -34,7 +34,10 @@ jobs:
       - name: Install dependencies & build
         run: |
           npm ci
+          npm pack
           cd tests
           npm ci
+          version=$(cat ../package.json | jq --raw-output .version)
+          npm install -D ../nextcloud-webpack-vue-config-$version.tgz
```
Comment (skjnldsv) [inline .github/workflows/npm-test.yml:41]:
```suggestion
          # Manually install dependency to check for incompatibilities
          version=$(cat ../package.json | jq --raw-output .version)
          npm install -D ../nextcloud-webpack-vue-config-$version.tgz
```

File: .github/workflows/npm-test.yml [hunk 2]
```diff
@@ -34,7 +34,10 @@ jobs:
       - name: Install dependencies & build
         run: |
           npm ci
+          npm pack
```
Comment (skjnldsv) [inline .github/workflows/npm-test.yml:37]:
```suggestion
          # Build local webpack-vue-config package
          npm ci
          npm pack
```

File: .github/workflows/npm-test.yml [hunk 3]
```diff
@@ -34,7 +34,10 @@ jobs:
...
           cd tests
           npm ci
+          version=$(cat ../package.json | jq --raw-output .version)
+          npm install -D ../nextcloud-webpack-vue-config-$version.tgz
           npm run dev
           npm run build
```
Comment (skjnldsv) [inline .github/workflows/npm-test.yml:43]:
```suggestion
      - name: Install webpack-vue-config dependencies & build
        run: |
          npm ci
          npm pack

      - name: Install dependencies & build test app
        run: |
          cd tests
          npm ci
          # Manually install dependency to check for incompatibilities
          version=$(cat ../package.json | jq --raw-output .version)
          npm install -D ../nextcloud-webpack-vue-config-$version.tgz
          # Build test app dev and production
          npm run dev
          npm run build
```

**Comment by skjnldsv** [comment]
Tests looks good @artonge, wanna merge  now?

**Comment by skjnldsv** [comment]
Release?

**Comment by artonge** [comment]
Major or minor ?
I would say minor, nothing is breaking change.

**Comment by skjnldsv** [comment]
Minor yep :)

**Pull request merged**